src/shared/codelens/csharpCodeLensProvider.ts

```diff
@@ -320,7 +325,36 @@ export function isPublicMethodSymbol(
         const signatureBeforeMethodNameRange = new vscode.Range(symbol.range.start, symbol.selectionRange.start)
         const signatureBeforeMethodName: string = document.getText(signatureBeforeMethodNameRange)
 
-        return REGEXP_RESERVED_WORD_PUBLIC.test(signatureBeforeMethodName)
+        if (REGEXP_RESERVED_WORD_PUBLIC.test(signatureBeforeMethodName)) {
+            return isValidMethodSignature(symbol)
+        }
+    }
+
+    return false
+}
+
+/**
+ * Returns whether or not a VS Code DocumentSymbol is a method that could be a Lambda handler
+ * * has one or more parameters
+ * * if there is more than one parameter, the second parameter is an ILambdaContext object
+ *   * does not check for extension/implementation of ILambdaContext
+ * @param symbol VS Code DocumentSymbol to analyze
+ */
+export function isValidMethodSignature(symbol: vscode.DocumentSymbol): boolean {
+    const parametersRegExp = /\(.*\)/
+    const lambdaContextType = 'ILambdaContext '
+
+    if (symbol.kind === vscode.SymbolKind.Method) {
+        const parametersArr = parametersRegExp.exec(symbol.name)
+        // reject if there are no parameters
+        if (!parametersArr) {
+            return false
+        }
+        // don't worry about removing parenthesis
+        const individualParams = parametersArr[0].split(',')
```

Comment:
1. This is probably a Good Enough implementation. Note that it will trip up on things like 
`public void Foo(SomeType<TIn, TOut> bar, ILambdaContext baz)`

1. Consider doing arr.split.map(trimWhitespace), and changing the lambdaContextType check to a "starts with" check

Reply:
> This is probably a Good Enough implementation. Note that it will trip up on things like
`public void Foo(SomeType<TIn, TOut> bar, ILambdaContext baz)`

I imagine that will be pretty common, good callout. I'll see what I can do (might have to be regex based).

> Consider doing arr.split.map(trimWhitespace), and changing the lambdaContextType check to a "starts with" check

Good point, can do. Now realizing this would pick up on `FakeILambdaContext foo` and `String thisIsMyILambdaContext `